File: Backend/Models/Usuario.cs
using Backend.Models.Enums;

namespace Backend.Models;

public class Usuario
{
    public int Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string SenhaHash { get; set; } = string.Empty;
    public PerfilUsuario Perfil { get; set; }
    public DateTime DataCriacao { get; set; } = DateTime.UtcNow;
    public bool Ativo { get; set; } = true;
}

File: backend/src/models/Usuario.ts
import { PerfilUsuario } from './enums';

/**
 * Interface que representa uma linha da tabela `usuarios`.
 * Nomes de propriedades = nomes das colunas no banco (snake_case).
 */
export interface Usuario {
  id: number;
  nome: string;
  email: string;
  senha_hash: string;
  perfil: PerfilUsuario;
  data_criacao: string;
  ativo: boolean;
}
